### PR TITLE
Add IOptions configuration to Compreface Console project

### PR DESCRIPTION
**Pull request by Asadbek07**

### CODE REVIEW

File: ComprefaceTestApp/Configuration/ComprefaceConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace ComprefaceTestApp.Configuration;
+
+public class ComprefaceConfiguration
```
Comment (yegorl) [inline ComprefaceTestApp/Configuration/ComprefaceConfiguration.cs:3]: Convert to interface and create 2 implementations - 1st one from config file (name should be provided). Probably passing IConfiguration will work as well i.e. to find a section etc. Raise exception if parameters are not set
2nd one should set parameters explicitly and injected as well.

File: ComprefaceTestApp/Program.cs [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 using ComprefaceTestApp.Services;
 using Flurl.Http;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Shared;
-using Shared.Constants;
 
 namespace ComprefaceTestApp;
```
Comment (yegorl) [inline ComprefaceTestApp/Program.cs:17]: Please create new class and interface IComprefaceClient. During startup all DI containers should raise up. Implement IDisposable. Program should looks like var client = new ComprefaceClient(apiKey, host); var result = client.CheckImage(img);

Comment (Asadbek07) [reply ComprefaceTestApp/Program.cs:17]: Done

File: ComprefaceTestApp/Configuration/ComprefaceConfiguration.cs (hunk 1 above)
Comment (Asadbek07) [reply ComprefaceTestApp/Configuration/ComprefaceConfiguration.cs]: Done

File: ComprefaceTestApp/Configuration/ComprefaceConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    
+    public ComprefaceConfiguration(string apiKey, string baseUrl)
+    {
+        BaseUrl = baseUrl ?? throw new ArgumentNullException($"{nameof(baseUrl)} cannot be null!");
+        ApiKey = apiKey ?? throw new ArgumentNullException($"{nameof(apiKey)} cannot be null!");
+    }
+
+    public ComprefaceConfiguration(IConfiguration configuration, string sectionForApiKey, string sectionForBaseUrl)
+    {
+        BaseUrl = configuration.GetSection(sectionForBaseUrl).Value ?? throw new ArgumentNullException($"Cannot read section: {sectionForBaseUrl} from the given configuration appsettings.json file");
```
Comment (Asadbek07) [inline ComprefaceTestApp/Configuration/ComprefaceConfiguration.cs:22]: default configuration file should be `appsettings.json`

File: ComprefaceTestApp/ComprefaceClient.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿using ComprefaceTestApp.Configuration;
+using ComprefaceTestApp.Services;
+using Microsoft.Extensions.Configuration;
+
+namespace ComprefaceTestApp;
```
Comment (yegorl) [inline ComprefaceTestApp/ComprefaceClient.cs:5]: Replace namespace to Exadel.Compreface

Comment (Asadbek07) [reply ComprefaceTestApp/ComprefaceClient.cs]: Done

**Review by yegorl — APPROVED** [review]

**Pull request merged**